model/task/task.go

```diff
@@ -675,17 +692,23 @@ func (t *Task) MarkStart(startTime time.Time) error {
 
 // SetResults sets the results of the task in TestResults
 func (t *Task) SetResults(results []TestResult) error {
-	t.TestResults = results
-	return UpdateOne(
-		bson.M{
-			IdKey: t.Id,
-		},
-		bson.M{
-			"$set": bson.M{
-				TestResultsKey: results,
-			},
-		},
-	)
+	catcher := grip.NewSimpleCatcher()
+	var testResult testresult.TestResult
+	for _, result := range results {
+		testResult = testresult.TestResult{
```

Comment:
It seems like there should be a method on testresult.TestResult or on task.TestResult which performs this conversion, which is a bit absurd 

It might be possible, if the structs have the same fields/layout to just coerce them directly, as well. 

I think this sort of illustrates how weird the models package hierarchy really is.

Reply:
done